test/registries.test.js

```diff
@@ -0,0 +1,48 @@
+describe('Registry', function () {
+  describe('one per app', function () {
+    it('should allow two apps to reuse the same model name', function (done) {
+      var appFoo = loopback();
+      var appBar = loopback();
+      var modelName = 'MyModel';
+      var subModelName = 'Sub' + modelName;
+      var settings = {base: 'PersistedModel'};
+      appFoo.set('perAppRegistries', true);
+      appBar.set('perAppRegistries', true);
```

Comment:
This flag does not seem to be implemented at all. I'd also prefer a different name. Since the flag is per app object, using plural name does not make sense to me.

I am proposing `app.set('localModelRegistry', true)` or `app.set('isolateModels', true)`.